posix/unix.c

```diff
@@ -592,7 +573,38 @@ int unix_getsockname(unsigned socket, struct sockaddr *address, socklen_t *addre
 
 int unix_getsockopt(unsigned socket, int level, int optname, void *optval, socklen_t *optlen)
 {
-	return 0;
+	unixsock_t *s;
+	int err;
```

Comment:
```suggestion
	int err = EOK;
```
and change only on error